feat(discover) Add tracing functions to column builder

markstory opened the pull request:
Add the rest of the tracing functions to the column builder modal. This also introduces multi-parameter functions to the UI and makes updates our parsing/serializing to handle it. I'm not happy with the `aggregation, field, refinement` structure and will refactor it into a simple array in a follow up pull request.

Some of the new functions only take a scalar argument, so I've used simple number inputs:

![Screen Shot 2020-03-06 at 4 08 42 PM](https://user-images.githubusercontent.com/24086/76123558-e3e7bd00-5fc6-11ea-96c0-d1e65f8900cf.png)

More complex functions take columns and parameters (refinements). When present these functions force the editor into a three column layout:

![Screen Shot 2020-03-06 at 4 05 44 PM](https://user-images.githubusercontent.com/24086/76123625-0da0e400-5fc7-11ea-86ad-b159ffa02917.png)

markstory committed:
feat(discover) Add tracing functions to column editor

Add expanded tracing functions to the column editor. The new functions
require a more complete description of the parameters and input types as
well as managing more input values. I've included required state, and
default values to start with so that using complex fields like apdex()
and impact() are more straightforward to use.

While I've added the idea of a 'refinement' to the EventView Field type,
I don't think it is a good idea long term. Instead I would like to
refactor the Field type to be `{field, parameters: []}` which should
help streamline other discover's guts.

markstory committed:
Update tests and make functions with no parameters not have ...

markstory committed:
Add autofocus and rudimentary focus styles.

markstory committed:
Don't always show three columns.

wmak reviewed (commented):
Not sure if issues with my local setup, but the percentile column is empty for me (even with trying existing percentiles like 95):
<img width="472" alt="image" src="https://user-images.githubusercontent.com/4205004/76125474-e2b88f00-5fca-11ea-8d9f-e9058a733e73.png">

And when trying to type in the function parameter the input field keeps kicking me out. (Tested both in Firefox + Chrome)
eg. if I Try to type `0.5` I can't, possibly cause of input validation?
Arrow inputs work though :+1:

markstory commented:
> And when trying to type in the function parameter the input field keeps kicking me out. (Tested both in Firefox + Chrome)
eg. if I Try to type 0.5 I can't, possibly cause of input validation?

I'll look into this. I was able to type into the filter boxes without problems. Too many `backspace` might send you away though.

markstory committed:
Fix alias keys being wrong.

markstory committed:
Fix input state

The onchange event fires on every keypress. Previously this would
re-render the entire row on each press which causes focus to be lost
making the fields frustrating to deal with. We need to handle state
closer to the source to prevent elements from being remounted. Having
a stateful input also enables validation rules to be enforced better.

markstory committed:
Use text inputs as number inputs are annoying.

markstory committed:
Fix typing issue.

markstory committed:
Fix tests.

markstory commented:
> And when trying to type in the function parameter the input field keeps kicking me out. (Tested both in Firefox + Chrome)

This was caused by how I was rendering the input boxes. It should be fixed now.

evanh commented on src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx:
```diff
@@ -100,6 +110,89 @@ export const AGGREGATIONS = {
+    isSortable: true,
+  },
+  rps: {
+    parameters: [
```
I don't understand fully how these are used throughout the frontend code, but RPM and RPS should not have a parameter that users can toggle. This value should only be used in the timeseries queries, to configure how many results are returned. If a user specifies a value for these in the main table results, it will mess up all the results.

markstory replied:
These definitions are used to build the input sets that we show users. If these are parameterless to the user, I'll remove these parameters.

evanh commented on tests/js/spec/views/eventsV2/results.spec.jsx:
```diff
@@ -16,7 +16,7 @@ const FIELDS = [
   },
   {
-    field: 'count(user)',
+    field: 'count(id)',
```
You got rid of `count(id)` everywhere else, why not here?

markstory replied:
Fixed now.

markstory committed:
Merge branch 'master' into feat/tracing-function-builder

markstory committed:
Remove parameters from rps() and rpm()

Users aren't supposed to set values on these.

markstory committed:
Fix missing props

markstory committed:
Remove more count(id)

markstory committed:
Ignore empty fields as they are bad.

dashed reviewed (approved):
Looks good to me 👍
